src/main/Application/Subscriptions/SubscriptionApplicationService.cs

```diff
@@ -1,33 +1,55 @@
 ﻿using ei8.Cortex.Diary.Domain.Model;
 using ei8.Cortex.Diary.Nucleus.Client.In;
 using ei8.Cortex.Subscriptions.Common;
+using ei8.Cortex.Subscriptions.Common.Receivers;
+using Microsoft.Extensions.DependencyInjection;
+using System;
 using System.Threading.Tasks;
 
 namespace ei8.Cortex.Diary.Application.Subscriptions
 {
     public class SubscriptionApplicationService : ISubscriptionApplicationService
     {
-		private readonly ISubscriptionClient subscriptionClient;
+		private readonly IServiceProvider services;
         private readonly ITokenManager tokenManager;
 
-        public SubscriptionApplicationService(ISubscriptionClient subscriptionClient, ITokenManager tokenManager)
+        public SubscriptionApplicationService(IServiceProvider services, ITokenManager tokenManager)
```

Comment:
Using `IServiceProvider` instead of the generic interface will allow us to resolve the specific instance needed for a particular subscription type (e.g. `SubscribeWithBrowserAsync` uses `ISubscriptionClient<BrowserReceiverInfo>`) Other methods in the service can resolve the required client as needed.

Reply:
I think either approach is correct. However, for consistency with other application services - it might be better to inject the client in the constructor:

"Constructor-based initialization is a worthy means to fulfill the dependencies, but they could have been retrieved by means of a Service Factory or using dependency injection. Your team standards and practices reign.
Vernon, Vaughn. Implementing Domain-Driven Design . Pearson Education. Kindle Edition. "

Reply:
Changed constructor to accept generic `ISubscriptionClient`